doc/rm/c_cpp_coding_style.md

```diff
@@ -257,6 +257,36 @@ The code using an X Macro must:
     It is usually clearer if you split the definition over multiple lines.
 *   Immediately after the definition, include the `.def` file for the X Macro.
 
+### Undefined Behavior
+
+***All uses of Undefined Behavior (unless stated otherwise) are forbidden without explicit approval of toolchain experts.***
```

Comment:
I don't think we have a good definition of 'toolchain experts'.

How about: `All uses of Undefined Behavior (unless stated otherwise) must be flagged and subjected to careful review.`

I wonder if we should also require some sort of explicit comment. For example, `// WARNING(UB): ...`. This would allow us to review uses of UB it in the future and make it less tempting to use undefined behavior to make code shorter.

Reply:
I prefer that phrasing - it helps make it clear that you can't really rely on a single person to always get this right. Discussions on llvm-dev show that even amongst expert toolchain developers there can be substantial confusion about various UB corner cases

Reply:
Yeah, I left this intentionally vague so that we can hash something out in the PR. I feel like it would be useful to have a dedicated mailing list for reviewing questions about all these "coding standards" questions, which is fairly cross-cutting. I'm going to change this into this text but leave an open for how to actually kick off review.